Makefile

```diff
@@ -23,3 +23,7 @@ integration-test: build
 
 integration-test-py3: build-py3
 	docker run -v `$(HOST_TMPDIR)`:/tmp -v /var/run/docker.sock:/var/run/docker.sock docker-py3 py.test -rxs tests/integration_test.py
+
+integration-ci:
+	docker build -t dpy-tests -f ./tests/Dockerfile .
+	docker run --privileged -t dpy-tests
```

Comment:
Instead of  running both processes in the same container, what do you think about doing this:

* change the base of this `Dockerfile` to be something like `debian:jessie`
* change the docker run link to the `dind` image instead

```
docker run -d --name dpy-dind --privileged dockerswarm/dind:1.8.1 docker -d  -H tcp://0.0.0.0:2375
docker run --rm --link=dpy-dind:docker --env="DOCKER_HOST=tcp://docker:2375"
docker rm -vf dpy-dind
```

(which would allow the `CMD` in the `Dockerfile` to change to just `py.test -rxs tests/integration_test.py`

`dpy-dind` could be a variable even, which would allow for jenkins to set a unique value for each build.

This is the approach aanand used in https://github.com/docker/compose/pull/1783